To: vim_dev@googlegroups.com
Subject: Patch 7.3.300
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.300
Problem:    Python doesn't parse multi-byte argument correctly.
Solution:   Use "t" instead of "s". (lilydjwg)
Files:	    src/if_py_both.h


*** ../vim-7.3.299/src/if_py_both.h	2011-07-07 15:08:53.000000000 +0200
--- src/if_py_both.h	2011-09-07 19:25:12.000000000 +0200
***************
*** 74,80 ****
      char *str = NULL;
      int error = ((OutputObject *)(self))->error;
  
!     if (!PyArg_ParseTuple(args, "es#", ENC_OPT, &str, &len))
  	return NULL;
  
      Py_BEGIN_ALLOW_THREADS
--- 74,80 ----
      char *str = NULL;
      int error = ((OutputObject *)(self))->error;
  
!     if (!PyArg_ParseTuple(args, "et#", ENC_OPT, &str, &len))
  	return NULL;
  
      Py_BEGIN_ALLOW_THREADS
***************
*** 114,120 ****
  	char *str = NULL;
  	PyInt len;
  
! 	if (!PyArg_Parse(line, "es#", ENC_OPT, &str, &len)) {
  	    PyErr_SetString(PyExc_TypeError, _("writelines() requires list of strings"));
  	    Py_DECREF(list);
  	    return NULL;
--- 114,120 ----
  	char *str = NULL;
  	PyInt len;
  
! 	if (!PyArg_Parse(line, "et#", ENC_OPT, &str, &len)) {
  	    PyErr_SetString(PyExc_TypeError, _("writelines() requires list of strings"));
  	    Py_DECREF(list);
  	    return NULL;
*** ../vim-7.3.299/src/version.c	2011-09-07 19:08:57.000000000 +0200
--- src/version.c	2011-09-07 19:26:05.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     300,
  /**/

-- 
A salesperson says:            Translation:
"backward compatible"          Old technology
"Premium"                      Overpriced
"Can't keep it on the shelf"   Unavailable
"Stands alone"                 Piece of shit
"Proprietary"                  Incompatible
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
